### PR TITLE
metamaps/mapDirectly

### PR DESCRIPTION
**Pull request by sofstam**
## PR checklist

This [PR](https://github.com/nf-core/modules/pull/1460) was stale because of a conda issue. The conda recipe has now been updated and this PR adds the module `metamaps/Mapdirectly`

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/metamaps/mapdirectly/environment.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/environment-schema.json
```
Comment (CarsonJM) [inline modules/nf-core/metamaps/mapdirectly/environment.yml:2]:
```suggestion
```

File: modules/nf-core/metamaps/mapdirectly/meta.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/meta-schema.json
```
Comment (CarsonJM) [inline modules/nf-core/metamaps/mapdirectly/meta.yml:2]:
```suggestion
```

File: modules/nf-core/metamaps/mapdirectly/main.nf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    input:
+    tuple val(meta), path(reads)
+    path database
+
+    output:
+    tuple val(meta), path("*classification_res")         , emit: classification_res
+    path("*classification_res.meta")                     , emit: meta_file
+    path("*classification_res.meta.unmappedReadsLengths"), emit: meta_unmappedreadsLengths
+    path("*classification_res.parameters")               , emit: para_file
```
Comment (CarsonJM) [inline modules/nf-core/metamaps/mapdirectly/main.nf:18]: Should these also have a meta associate with them?

File: modules/nf-core/metamaps/mapdirectly/main.nf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        --threads $task.cpus \\
+        --query $reads \\
+        --output ${prefix}.classification_res
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        metamaps: \$(metamaps | sed -n 2p | sed 's/^.*MetaMaps v //')
+    END_VERSIONS
+    """
+}
```
Comment (CarsonJM) [inline modules/nf-core/metamaps/mapdirectly/main.nf:42]:
```suggestion
    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_classification_res
    touch ${prefix}_classification_res.meta
    touch ${prefix}_classification_res.meta.unmappedReadsLengths
    touch ${prefix}_classification_res.parameters
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metamaps: \$(metamaps | sed -n 2p | sed 's/^.*MetaMaps v //')
    END_VERSIONS
    """
}
```

File: modules/nf-core/metamaps/mapdirectly/tests/nextflow.config [hunk 5]
```diff
@@ -0,0 +1,5 @@
+process {
+    withName: UNTAR {
+        ext.args2 = "--no-same-owner"
```
Comment (CarsonJM) [inline modules/nf-core/metamaps/mapdirectly/tests/nextflow.config:3]: Just out of curiosity, why is this flag being added?

**Review by CarsonJM — CHANGES_REQUESTED** [review]
Looks good overall! Just a couple comments/questions!

File: modules/nf-core/metamaps/mapdirectly/tests/nextflow.config (hunk 5 above)
Comment (CarsonJM) [reply modules/nf-core/metamaps/mapdirectly/tests/nextflow.config:3]: Good to know, I'll keep this in mind if I come across this in the future 👍

**Comment by CarsonJM** [comment]
@nf-core-bot Fix linting

**Review by CarsonJM — APPROVED** [review]
@sofstam As long as you think the other outputs don't need a meta this looks good to me!

File: modules/nf-core/metamaps/mapdirectly/main.nf (hunk 3 above)
Comment (sofstam) [reply modules/nf-core/metamaps/mapdirectly/main.nf]: I have added the meta now.

**Comment by sofstam** [comment]
I updated the PR with the meta for the other outputs.

**Review by CarsonJM — APPROVED** [review]
LGTM! Thanks for making these changes!

**Pull request merged**